go/types/collection.go

```diff
@@ -8,5 +8,6 @@ type Collection interface {
 	Value
 	Len() uint64
 	Empty() bool
+	IsLeaf() bool
```

Comment:
yeah, the other option was moving this into `types`. I figured you'd prefer this.